ValheimPlus/GameClasses/EnvMan.cs

```diff
@@ -6,65 +6,90 @@ namespace ValheimPlus.GameClasses
 {
     public class TimeManipulation
     {
-        /*
         /// <summary>
-        /// Alter the total day length in seconds base on configuration file
+        /// Hook on EnvMan init to alter total day length
         /// </summary>
-        public static void SetupDayLength()
+        [HarmonyPatch(typeof(EnvMan), nameof(EnvMan.Awake))]
+        public static class TimeInitHook
         {
-            if (Configuration.Current.Time.IsEnabled)
+            private static void Prefix(ref EnvMan __instance)
             {
-                EnvMan instance = EnvMan.m_instance;
-                if (instance)
-                {
-                    instance.m_dayLengthSec = (long)Configuration.Current.Time.totalDayTimeInSeconds;
-                }
-                else
+                if (Configuration.Current.Time.IsEnabled)
                 {
-                    ValheimPlusPlugin.logger.LogWarning("EnvMan instance not loaded");
+                    if (__instance)
+                    {
+                        __instance.m_dayLengthSec = (long)Configuration.Current.Time.totalDayTimeInSeconds;
+                    }
+                    else
+                    {
+                        ValheimPlusPlugin.Logger.LogWarning("EnvMan instance not loaded");
+                    }
                 }
             }
         }
 
         /// <summary>
-        /// Hook on EnvMan init to alter total day length
+        /// Hook on EnvMan RescaleDayFraction to modify the time of day we are currently in.
         /// </summary>
-        [HarmonyPatch(typeof(EnvMan), "Awake")]
-        public static class TimeInitHook
+        [HarmonyPatch(typeof(EnvMan), nameof(EnvMan.RescaleDayFraction))]
+        public static class RescaleDayFractionPatch
```

Comment:
Patch class rename
```suggestion
        public static class EnvMan_RescaleDayFraction_Patch
```
Although it may be a transpile soon.